packages/astro/src/core/build/index.ts

```diff
@@ -85,6 +85,19 @@ class AstroBuilder {
 			routeCache: this.routeCache,
 			viteServer: this.viteServer,
 		});
+
+		// Filter pages by using conditions based on their frontmatter.
+		Object.entries(allPages).forEach(([page, data]) => {
+			if ('frontmatter' in data.preload[1]) {
+				// TODO: add better type inference to data.preload[1]
```

Comment:
nit: for a followup PR that I will make.

We do not have consistent formatting for TODOs, which makes finding them a little less obvious.

For a loose check of "todo", I found 360 of these in our project. We will want to audit these at some point.